app/services/support_interface/ministerial_report_candidates_export.rb

```diff
@@ -0,0 +1,118 @@
+    end
+
+    def call
+      export_rows = {}
+
+      MinisterialReport::SUBJECTS.each { |subject| export_rows[subject] = column_names }
+
+      export_rows[:split] = column_names
+
+      application_forms.each do |application|
+        subjects = determine_subjects(application)
+
+        states = determine_states(application)
+
+        if candidate_has_no_dominant_subject?(subjects)
+          states.each { |state| export_rows[:split][state] += 1 }
+        else
+          dominant_subject = dominant_subject(subjects)
+          next if dominant_subject.nil?
+
+          states&.each { |state| add_row_values(export_rows, dominant_subject, state) }
+        end
+      end
+
+      export_rows[:total] = export_rows[:primary].merge(export_rows[:secondary]) { |_k, primary_value, secondary_value| primary_value + secondary_value }
+
+      export_rows.map { |subject, value| { subject: subject }.merge!(value) }
+    end
+
+    alias data_for_export call
+
+    def determine_states(application)
+      choice_statuses = application.application_choices.map(&:status)
+
+      if choice_statuses.any? { |choice_status| ApplicationStateChange::SUCCESSFUL_STATES.include? choice_status.to_sym }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:recruited]
+      elsif choice_statuses.any? { |choice_status| %w[offer conditions_not_met].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:offer]
+      elsif choice_statuses.any? { |choice_status| %w[awaiting_provider_decision interviewing].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:awaiting_provider_decision]
+      elsif choice_statuses.any? { |choice_status| %w[offer_withdrawn].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:offer_withdrawn]
+      elsif choice_statuses.any? { |choice_status| %w[cancelled declined].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:declined]
+      elsif choice_statuses.any? { |choice_status| %w[rejected].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:rejected]
+      elsif choice_statuses.any? { |choice_status| %w[withdrawn].include? choice_status }
+        MinisterialReport::CANDIDATES_REPORT_STATUS_MAPPING[:withdrawn]
+      end
+    end
+
+  private
+
+    def add_row_values(hash, subject, state)
+      hash[:stem][state] += 1 if MinisterialReport::STEM_SUBJECTS.include? subject
+      hash[:ebacc][state] += 1 if MinisterialReport::EBACC_SUBJECTS.include? subject
+      hash[:secondary][state] += 1 if MinisterialReport::SECONDARY_SUBJECTS.include? subject
+      hash[subject][state] += 1
+    end
+
+    def determine_subjects(application_form)
+      courses = application_form.application_choices.map(&:course)
+      courses.map { |course| dominant_subject_for_course(course) }
```

Comment:
I guess this could contain `nil`s in the event the substring matching failed — which could mean `candidate_has_no_dominant_subject([nil, nil])`, which would mean that candidate supposedly has a dominant subject, though we don't know what it is?

Reply:
would be good to identify the subject but if this scenario is true, as a fallback, allocate them to the primary/secondary subtotal row as previously discussed?

Reply:
yes